app/models/user.rb

```diff
@@ -72,32 +72,33 @@ class User < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :created_podcasts, class_name: "Podcast", foreign_key: :creator_id, inverse_of: :creator, dependent: :nullify
   has_many :credits, dependent: :destroy
-  has_many :discussion_locks, dependent: :destroy, inverse_of: :locking_user, foreign_key: :locking_user_id
-  has_many :display_ad_events, dependent: :destroy
+  has_many :discussion_locks, dependent: :delete_all, inverse_of: :locking_user, foreign_key: :locking_user_id
+  has_many :display_ad_events, dependent: :delete_all
   has_many :email_authorizations, dependent: :delete_all
   has_many :email_messages, class_name: "Ahoy::Message", dependent: :destroy
   has_many :field_test_memberships, class_name: "FieldTest::Membership", as: :participant, dependent: :destroy
+  # Consider that we might be able to use dependent: :delete_all as the GithubRepo busts the user cache
   has_many :github_repos, dependent: :destroy
   has_many :html_variants, dependent: :destroy
-  has_many :identities, dependent: :destroy
+  has_many :identities, dependent: :delete_all
   has_many :identities_enabled, -> { enabled }, class_name: "Identity", inverse_of: false
   has_many :listings, dependent: :destroy
-  has_many :mentions, dependent: :destroy
-  has_many :notes, as: :noteable, inverse_of: :noteable, dependent: :destroy
-  has_many :notification_subscriptions, dependent: :destroy
-  has_many :notifications, dependent: :destroy
+  has_many :mentions, dependent: :delete_all
+  has_many :notes, as: :noteable, inverse_of: :noteable, dependent: :delete_all
```

Comment:
This one actually sets the `author_id` to `NULL` instead of cascading the deletion. Not sure why.

```
"fk_rails_36c9deba43" FOREIGN KEY (author_id) REFERENCES users(id) ON DELETE SET NULL
```